contracts/ValidGroupStorage.sol

```diff
@@ -0,0 +1,36 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+import "@interep/contracts/Interep.sol";
+
+contract ValidGroupStorage {
+    mapping(uint256 => bool) public validGroups;
+
+    Interep public interep;
+
+    struct Group {
```

Comment:
Note that this is a partial struct of Interep's [Group Struct](https://github.com/interep-project/contracts/blob/24c9db8efb7ad28b3e5dc9659f0a5c1cb6e63add/contracts/IInterep.sol#L12-L16) because from the RLN perspective, we're only concerned with provider+tier pairings, i.e GitHub gold, GitHub silver, etc. There is validation in the constructor to ensure that the group actually exists on Interep's contract.